apm-agent-plugin-sdk/src/main/java/co/elastic/apm/agent/sdk/weakmap/NullCheck.java

```diff
@@ -0,0 +1,38 @@
+package co.elastic.apm.agent.sdk.weakmap;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nullable;
+
+public class NullCheck {
+
+    private static final Logger logger = LoggerFactory.getLogger(NullCheck.class);
+
+    /**
+     * checks if key or value is {@literal null}
+     *
+     * @param v key or value
+     * @return {@literal true} if key is non-null, {@literal false} if null
+     */
+    private static <T> boolean isNull(@Nullable T v, boolean isKey) {
+        if (null != v) {
+            return false;
+        }
+        String msg = String.format("trying to use null %s", isKey ? "key" : "value");
+        if (logger.isDebugEnabled()) {
+            logger.debug(msg, new RuntimeException(msg));
+        } else {
+            logger.warn(msg);
+        }
+        return true;
+    }
+
+    public static <T> boolean isNullKey(@Nullable T key){
+        return true;
+    }
+
+    public static <T> boolean isNullValue(@Nullable T value) {
+        return true;
+    }
```

Comment:
❓

Reply:
sorry, forgot to update those, it's not really surprising that test fails with that :-)